test/parallel_api/algorithm/alg.sorting/sort.pass.cpp

```diff
@@ -177,12 +179,62 @@ Equal(int32_t x, int32_t y)
 template <typename T>
 struct test_sort_with_compare
 {
+#if TEST_DPCPP_BACKEND_PRESENT
+#if _PSTL_SYCL_TEST_USM
+    template <sycl::usm::alloc alloc_type, typename Policy, typename InputIterator, typename OutputIterator,
+              typename OutputIterator2, typename Size, typename Compare>
+    typename ::std::enable_if<is_base_of_iterator_category<::std::random_access_iterator_tag, InputIterator>::value,
+                              void>::type
+    test_usm(Policy&& exec, OutputIterator tmp_first, OutputIterator tmp_last, OutputIterator2 expected_first,
+             OutputIterator2 expected_last, InputIterator first, InputIterator /* last */, Size n, Compare compare)
+    {
+        using namespace std;
+        copy_n(first, n, expected_first);
+        copy_n(first, n, tmp_first);
+        if (Stable)
+            ::std::stable_sort(expected_first + 1, expected_last - 1, compare);
+        else
+            ::std::sort(expected_first + 1, expected_last - 1, compare);
+
+        const auto itSortFrom = tmp_first + 1;
+        const auto itSortTo = tmp_last - 1;
+        const auto size = itSortTo - itSortFrom;
+
+        using _src_type = decltype(*itSortFrom);
+        using _src_type_decay = std::decay_t<_src_type>;
+
+        auto queue = exec.queue();
+
+        // allocate USM memory and copying data to USM shared/device memory
+        TestUtils::usm_data_transfer<alloc_type, _src_type_decay> dt_helper(queue, itSortFrom, itSortTo);
+        auto sortingData = dt_helper.get_data();
+
+        int32_t count0 = KeyCount;
+        if (Stable)
+            stable_sort(exec, sortingData, sortingData + size, compare);
+        else
+            sort(exec, sortingData, sortingData + size, compare);
+
+        // check result
+        dt_helper.retrieve_data(::std::addressof(*itSortFrom));
+
+        for (size_t i = 0; i < n; ++i, ++expected_first, ++tmp_first)
```

Comment:
There is `EXPECT_EQ_N` in the test utilities - see utils.h

Reply:
Fixed